pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -367,7 +368,9 @@ func NewNodeLifecycleController(
 ) (*Controller, error) {
 
 	if kubeClient == nil {
-		klog.Fatalf("kubeClient is nil when starting Controller")
+		klog.ErrorS(nil, "kubeClient is nil when starting Controller")
+		// ACTION REQUIRED: Exit code changed from 255 to 1
+		os.Exit(1)
```

Comment:
`klog.FlushAndExit(1)`

Reply:
Or return an error? This function can do that. I haven't checked the others.

Reply:
yeah, maybe return a error is better here.